docarray/array/mixins/io/binary.py

```diff
@@ -73,6 +73,45 @@ def load_binary(
                     for od in track(d[_len:].split(_binary_delimiter))
                 )
 
+
+    @classmethod
+    def load_binary_stream(
+        cls: Type['T'],
+        file: Union[str, BinaryIO, bytes],
+    ) -> 'T':
+
+        if os.path.exists(file):
+            file_ctx = open(file, 'rb')
+        else:
+            raise ValueError(f'unsupported input {file!r}')
+
+        from .... import Document
+
+        delimiter = None
+        current_bytes = b''
+        with file_ctx as fp:
+            while True:
+                b = current_bytes + fp.read(500)
+                if delimiter is None:
+                    _len = len(random_uuid().bytes)
```

Comment:
I think for efficiency and mantainability, this method should rely on using `load_binary` from DocumentArray itself

Reply:
1. 
```
if delimiter is None:
                    _len = len(random_uuid().bytes)
```
this should be move out of the loop and the first `_len` chars are always **the** delimiter. 

2. make sure to cover `compress` as otherwise the usage is limited. Note, all compressions I used there they **already** implemented streamed file handler, you just `open` it, check Python official docs & lz4 docs for details. Refer to other part of the code on how I implement compression.
3. Agree with Joan, `load_binary(..., stream: bool = False) -> Union[DocumentArray, Generator[DocumentArray, ...]]`  is better. Note my type hint syntax on `Generator` may not be correct.